aptos-move/e2e-move-tests/src/harness.rs

```diff
@@ -357,6 +362,58 @@ impl MoveHarness {
         )
     }
 
+    /// Creates a transaction which publishes the Move Package found at the given path on behalf
+    /// of the given account.
+    ///
+    /// The passed function allows to manipulate the generated metadata for testing purposes.
+    pub fn create_publish_package(
+        &mut self,
+        account: &Account,
+        path: &Path,
+        options: Option<BuildOptions>,
+        patch_metadata: impl FnMut(&mut PackageMetadata),
+    ) -> SignedTransaction {
+        let package = BuiltPackage::build(path.to_owned(), options.unwrap_or_default())
+            .expect("building package must succeed");
+        self.create_publish_built_package(account, &package, patch_metadata)
+    }
+
+    pub fn create_publish_package_cache_building(
+        &mut self,
+        account: &Account,
+        path: &Path,
+        patch_metadata: impl FnMut(&mut PackageMetadata),
+    ) -> SignedTransaction {
+        let package_arc = {
+            let mut cache = CACHED_BUILT_PACKAGES.lock().unwrap();
+
+            cache
+                .entry(path.to_owned())
+                .or_insert_with(|| {
+                    Arc::new(BuiltPackage::build(
+                        path.to_owned(),
+                        BuildOptions::default(),
+                    ))
+                })
+                .clone()
```

Comment:
Here, does the `clone` actually clone the package data, or does the `clone` simply creates a new pointer for the package data?

Reply:
it clones the Arc

Reply:
maybe Arc::clone to avoid confusion